multiversx_sdk/network_providers/errors.py

```diff
@@ -21,3 +21,5 @@ def __init__(self) -> None:
 class TransactionFetchingError(Exception):
     def __init__(self, url: str, error: Any):
         super().__init__(f"Couldn't fetch transaction on url = [{url}], error = [{error}]")
+        self.url = url
```

Comment:
Idea: maybe also derive from `NetworkProviderError`? So that client code can handle `NetworkProviderError`(s) homogeneously?

Reply:
I think it would be nice